whistl/util.py

```diff
@@ -0,0 +1,126 @@
+'''This file contains useful functions for processing data'''
+
+
+def get_labels(df, sample_to_label, label_to_encoding):
+    ''' Retrieve the labels for the given dataset
+
+    Arguments
+    ---------
+    df: pandas.DataFrame
+        The DataFrame to generate labels for. Each column in the DataFrame should be a
+        sample contained in sample_to_label
+    sample_to_label: dict
+        A dictionary mapping sample ids to their label
+    label_to_encoding: dict
+        A dictionary mapping the string version of a label e.g. 'sepsis' to the int encoded version
+        e.g. 0
+
+    Returns
+    ------
+    labels: list of ints
+        The labels to be used in training a model
+    '''
+    labels = []
+    for column in df.columns:
+        labels.append(label_to_encoding[sample_to_label[column]])
+
+    return labels
+
+
+def remove_samples_with_label(df, sample_to_label, label_to_remove):
+    ''' Remove all samples with a given label from the DataFrame
+
+    Arguments
+    ---------
+    df: pandas.DataFrame
+        The DataFrame to be filtered. Each column in the DataFrame should be a
+        sample contained in sample_to_label
+    sample_to_label: dict
+        A dictionary mapping sample ids to their label
+    label_to_remove: str
+        The name of the label to remove. For example, 'other'
+
+    Returns
+    -------
+    df: pandas.DataFrame
+        The filtered version of the dataframe passed in
+    '''
+    keep_columns = [col for col in df.columns if sample_to_label[col] != label_to_remove]
+    df = df[keep_columns]
+
+    return df
+
+
+def parse_gene_file(gene_file_path):
+    '''Read a list of genes from a file
+
+    Arguments
+    ---------
+    gene_file_path: str
+        The path to the csv file to be read from
+
+    Returns
+    -------
+    genes: list of str
+        The genes found in the file
+    '''
+    with open(gene_file_path, 'r') as gene_file:
+        genes = []
+        for line in gene_file:
+            genes.append(line.strip().strip(','))
+
+    return genes
+
+
+def get_class_weights(train_loader):
+    '''Calculate class weights for better training performance on unbalanced data
+
+    Arguments
+    ---------
+    train_loader: pytorch.DataLoader
+        The data loader for a ExpressionDataset containing the labels to calculate weights for
+
+    Returns
+    -------
+    weights: dict
+        A dictionary mapping encoded labels to weights
+    '''
+    value_counts, total = get_value_counts(train_loader)
+
+    # Weight labels according to the inverse of their frequency
```

Comment:
Out of curiosity, how imbalanced are the labels for this dataset from what you've seen?

Reply:
It's roughly a 90/10 split of sepsis:healthy expression